internal/wrappers/features/documents.feature

```diff
@@ -1,17 +1,210 @@
+    When I createOrReplace a document with id 'my-document-id'
+    Then the document is successfully created
+
+  Scenario: Replace a document if it exists
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    When I createOrReplace a document with id 'my-document-id'
+    Then the document is successfully replaced
+
+  Scenario: Replace a document
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'replace-my-document-id'
+    When I replace a document with id 'replace-my-document-id'
+    Then the document is successfully replaced
+
+  Scenario: Delete a document
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'delete-my-document-id'
+    When I delete the document with id 'delete-my-document-id'
+    Then the document is successfully deleted
+
+  Scenario: Update a document
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'update-my-document-id'
+    When I update a document with id 'update-my-document-id'
+    Then the document is successfully updated
+
+  Scenario: Search a document by id and find it
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'search-my-document-id'
+    When I search a document with id 'search-my-document-id'
+    Then the document is successfully found
+
+  Scenario: Search a document by id and don't find it
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'search-my-document-id'
+    When I search a document with id 'fake-id'
+    Then the document is not found
+
+  Scenario: Count documents in a collection
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'count-test-collection'
+    And the collection has a document with id 'count-my-document-id'
+    And the collection has a document with id 'count-my-document-id2'
+    And the collection has a document with id 'count-my-document-id3'
+    When I count how many documents there is in the collection
+    Then I shall receive 3
+
+  Scenario: Delete multiple documents with no error
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'delete-test-collection'
+    And the collection has a document with id 'mdelete-my-document-id'
+    And the collection has a document with id 'mdelete-my-document-id2'
+    When I delete the documents 'mdelete-my-document-id' and 'mdelete-my-document-id2'
```

Comment:
In order to better see that we are working on document lists, the step could be written with actual list:
And I delete the following documents: [ 'mdelete-my-document-id', 'mdelete-my-document-id2' ]